### DISCUSSION TITLE
Help accesing realtime database and getting data to dictionary

### DISCUSSION THREAD

**Discussion by Kontrano** [GodotNuts/GodotFirebase#285] [Q&A] [answered]
So im trying to use the real-time database from Firebase as a database to store event information and use this data to run certain functions in my Godot project. 

I've looked at the documentation and the chat example using the realtime database but i can't figure out how to retrieve the information? I have gotten authentication to work just fine though.

Could someone help me get access to the data or explain in more detail how to do so? I have included my relevant code below and an image of the database, How would i go about creating a dictionary for every Event or just the entire event tree inside my project from the database. Currently, I receive no response from the database at all. 

Help is most appreciated :)
![image](https://user-images.githubusercontent.com/80490564/158996097-528ae0d3-0961-4a67-9eab-7503af1e17b8.png)

 ```
 func _on_FirebaseAuth_login_succeeded(auth_info):
	print("logged in")
	userinfo = auth_info
	Global.LogedIn = true
	$Interface_Login.hide()
	$EventsOverview.show()
	EventListener = Firebase.Database.get_database_reference("dbei-plugin-default-rtdb",{})
	EventListener.connect("patch_data_update",self,"on_received_updated_event")
	EventListener.connect("patch_new_update",self,"on_received_new_event")
	
	
func on_received_new_event(data):
	print(2)
	if data.data:
		add_event(data.data.Time,data.data.Speaker)
		print(data.data.time)
func on_received_updated_event(data):
	print(data)
	print(1)
	
func add_event(time,speaker):
	pass
```

**Comment by WolfgangSenff** [comment, marked as answer]
You have an incorrect signal listed for the second one. The two signals you want are:
patch_data_update
and
new_data_update

You're using `patch_new_update`, which isn't a signal we have. You'll be able to tell by the error in Godot when attempting to run that line - it'll say the signal doesn't exist so cannot connect to it.

See more at these places:

https://github.com/GodotNuts/GodotFirebase/blob/main/addons/godot-firebase/database/reference.gd#L9
https://github.com/GodotNuts/GodotFirebase/wiki/Realtime-Database#firebasedatabasereference

**Reply by Kontrano** [reply]
Hmm it didn't throw me any errors while running, but thanks ill change it and attempt again. I seem to have mistyped and kept reading over it.

**Reply by Kontrano** [reply]
Now it works like a charm! I do seem to have an issue with the patch_data_update, whenever I change data no signal is sent back to the client or am i misunderstanding the implementation for this?

**Reply by WolfgangSenff** [reply]
Where are you changing it? If you're changing it in the console, this is a known issue with Firebase and they have no plans to fix it. :/ If you change it from another instance of your game, it should work.

**Reply by Kontrano** [reply]
Ah... yeah, I was changing it from the console, this is slightly annoying for my use case but ill find a way around it, Is there a way to do a manual refresh of the database reference?

**Reply by WolfgangSenff** [reply]
You could certainly just recreate the reference to it if you wanted. It isn't cached.

**Reply by Kontrano** [reply]
Thanks that worked perfectly, guess'll implement it this way for the current stage, thanks a lot for your help and quick response!

For anyone else reading this, you have to redo the database reference as well as the signal linking then it works.
